### ISSUE TITLE
[libmysql] build failure in libedit because warnings as errors

### ISSUE THREAD

**Issue by silverqx** [microsoft/vcpkg#39187]
### Operating system

Fedora 40

### Compiler

gcc 14.1

### Steps to reproduce the behavior

```Shell
vcpkg install libmysql
```

### Failure logs

[install-x64-linux-dbg-out.log](https://github.com/user-attachments/files/15743950/install-x64-linux-dbg-out.log)

### Additional context

I'm trying to port my GH actions to the latest `Fedora` as it has latest compilers and I can't compile `libmysql` because of **warnings as errors** are enabled in `libedit`? It looks like that I'm not exactly sure.

I found a source of this problem, starting described [here](https://lwn.net/Articles/972683/) and in depth [here](https://lwn.net/Articles/913505/) and [here](https://lwn.net/Articles/954018/).

The reason for this is: "It's -Werror=implicit-int -Werror=implicit-function-declaration. Best to throw in -Werror=int-conversion -Werror=strict-prototypes -Werror=old-style-definition."

Also, I would like to ask if we could update `libmysql` to the latest version, there is already `v8.0.37` with a few important [Auth-related fixes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-35.html#mysqld-8-0-35-capi)

OT: qtbase[core]:x64-linux/-dynamic compiles fine.

**Comment by silverqx** [comment]
[Here](https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors) is GCC's part describing porting to GCC 14.

I was able to compile it using:
```
CFLAGS="-Wno-implicit-function-declaration -Wno-int-conversion -Wno-error=implicit-int -Wno-error=implicit-function-declaration" vcpkg install libmysql
```

**Comment by MonicaLiu0311** [comment]
Can you try `-Wno-incompatible-pointer-types`? Refer to [39263/files](https://github.com/microsoft/vcpkg/pull/39263/files).

**Comment by dg0yt** [comment]
Or you make a custom triplet which injects `-fpermissive` for gcc-14?
https://gcc.gnu.org/gcc-14/porting_to.html#c

**Comment by silverqx** [comment]
Fixed by #40373
